src/coreclr/jit/codegencommon.cpp

```diff
@@ -5552,6 +5555,13 @@ void CodeGen::genFnProlog()
         genEnregisterIncomingStackArgs();
     }
 
+#ifdef TARGET_RISCV64
+    if (compiler->fgFirstBBisScratch() && compiler->lvaHasAnyStackParamToReassemble())
+    {
+        genHomeStackParams(/* handleStack */ true);
+    }
+#endif
```

Comment:
The scratch BB is not necessary when it happens during the prolog.

I think it would be better to do this before you home register parameters (in the non-OSR case above), and passing `initReg`/`initRegZeroed` as well so there is no chance of overwriting a homed register.